### PR TITLE
Individual state tests and adding Tomcats SimpleTcpCluster support.

### PR DESCRIPTION
**Pull request by fintanmm**
Hi all,

There is a lot in this PR. The commits are a bit off as I rebased twice. So apologise for that in advance. I have created tests for each state because some states were failing when they were called by themselves. 

I commented out arch testing as it was failing on every test with unsupported platform being the cause reported by chef. 

I have added an additional check to see if Tomcat is actually running by checking catalina.out. I also added rudimentary support for simple Tomcat clustering.  

Please let me know what you think. 

Regards,
Fintan

### CODE REVIEW

**Comment by gravyboat** [comment]
Can you rebase again? There were some changes I merged to the test file and osmap that might be a conflict. @fortune-juggle can you review this as well since it seems like there are some conflicts with the PR I merged from you?

File: tomcat/native.sls [hunk 1]
```diff
@@ -1,5 +1,8 @@
 {% from "tomcat/map.jinja" import tomcat with context %}
 
+include:
+  - tomcat.config
```
Comment (fortune-juggle) [inline tomcat/native.sls]: :+1:

File: tomcat/osmap.yaml [hunk 2]
```diff
@@ -13,7 +13,7 @@ Debian:
   service: tomcat8
   user: tomcat8
   group: tomcat8
-  java_home: /usr/lib/jvm/java-7-openjdk
+  java_home: /usr/lib/jvm/java-7-openjdk-amd64
```
Comment (fortune-juggle) [inline tomcat/osmap.yaml]: i would change this to defaul-java (it is a symlink to the right architecture)

**Comment by fortune-juggle** [comment]
Hi @gravyboat this looks good to me. No trouble to rebase with my changes . But after all I have to add my changes to all new tests.

**Comment by fintanmm** [comment]
No problem @gravyboat I'll have a look at it today.

**Comment by fintanmm** [comment]
Hi @gravyboat, @fortune-juggle 

I've done the rebase again. Can someone please review the changes? Thanks

**Comment by fortune-juggle** [comment]
Hi @fintanmm looks good. After merge I have to add my changes to the other tests

**Pull request merged**